test/stressHistogram.cxx

```diff
@@ -7030,44 +7030,33 @@ int findBin(ROOT::Fit::BinData& bd, const double *x)
    return -1;
 }
 
-bool operator ==(ROOT::Fit::BinData& bd1, ROOT::Fit::BinData& bd2)
+bool operator==(ROOT::Fit::BinData &bd1, ROOT::Fit::BinData &bd2)
 {
    const unsigned int ndim = bd1.NDim();
    const unsigned int npoints = bd1.NPoints();
+   const double eps = TMath::Limits<double>::Epsilon();
 
-   bool equals = true;
-
-   for ( unsigned int i = 0; i < npoints && equals; ++i )
-   {
+   for (unsigned int i = 0; i < npoints; ++i) {
       double value1 = 0, error1 = 0;
       const double *x1 = bd1.GetPoint(i, value1, error1);
 
-//       std::cout << "i: " << i
-//            << " x: ";
-//       std::copy(x1, x1+ndim, ostream_iterator<double>(std::cout, " "));
-//       std::cout << " val: " << value1
-//            << " error: " << error1
-//            << std::endl;
-
       int bin = findBin(bd2, x1);
-      if ( bin < 0 )
-         Fatal("operator ==(ROOT::Fit::BinData& bd1, ROOT::Fit::BinData& bd2)","BIN NOT FOUND!");
 
       double value2 = 0, error2 = 0;
       const double *x2 = bd2.GetPoint(bin, value2, error2);
 
-      equals &= ( value1 == value2 );
-      equals &= ( error1 == error2 );
-      for ( unsigned int j = 0; j < ndim; ++j )
-      {
-         equals &= fabs(x1[j] - x2[j]) < 1E-15;
+      if (value1 != value2) return false;
```

Comment:
Good point. I though value was an integer from the output of my debug statements, but looking at the interface it is indeed floating point. I will update it.